eZ/Publish/Core/MVC/Symfony/FieldType/RichText/Renderer.php

```diff
@@ -366,17 +383,10 @@ protected function getEmbedTemplateName($resourceType, $isInline, $isDenied)
      *
      * @throws \Symfony\Component\Security\Core\Exception\AccessDeniedException
      *
-     * @param int|string $id
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
      */
-    protected function checkContent($id)
+    protected function checkContent($content)
```

Comment:
I guess that type change of argument in a protected method is BC break, am I right?

Reply:
@glye ? :)

Reply:
Yes, not sure about exact BC rules but does sound dubious. Could be reverted, if we either accept that content is fetched twice (bad) or refactor some other way (possible BC problem again).

Reply:
You can add method params and stay BC, right? So add $content as second param, default null, use it instead of $id if not null.

Reply:
cleanest is to create new method, that takes content with typehint and everything, and reuse it in checkContent which you deprecate.

Reply:
Hmm, it's protected method inside of a Core class, so it should be fine even to do:
`protected function checkContent(Content $content)`
and refactor usages.
Unless it's used by external packages somehow? We're breaking API/SPI rules all over the place, so I'm not really sure. If that's the case then what @andrerom suggests is really the safest approach.

Reply:
Ideally we should not break those rules, and as this is backport, lets play it safe.